src/helm/benchmark/annotation/model_as_judge_annotator.py

```diff
@@ -0,0 +1,46 @@
+import json
+import datasets
+import os
+import re
+from typing import Dict
+
+from helm.common.general import ensure_directory_exists
+from helm.benchmark.adaptation.request_state import RequestState
+from helm.benchmark.annotation.annotator import Annotator
+from helm.clients.auto_client import AutoClient
+from helm.common.request import Request
+
+
+class ModelAsJudgeAnnotator(Annotator):
+    """Model as judge autograder."""
+
+    # expects a prompt that includes a user request that ensures the model needs to return a score and reasoning
+    def autograde_with_reasoning(
```

Comment:
nit: `score_with_reasoning` seems more consistent since we don't use the term "grade"